mlrun/artifacts/base.py

```diff
@@ -1276,18 +1281,35 @@ def get_artifact_meta(artifact):
 
 
 def generate_target_path(item: Artifact, artifact_path, producer):
-    # path convention: artifact_path[/{run_name}]/{iter}/{key}.{suffix}
+    # path convention: artifact_path[/{run_name}]/{iter}/src_path/{key}.{suffix}
     # todo: add run_id here (vs in the .run() methods), support items dedup (by hash)
     artifact_path = artifact_path or ""
-    if artifact_path and not artifact_path.endswith("/"):
-        artifact_path += "/"
+    if artifact_path and not artifact_path.endswith(os.sep):
```

Comment:
not sure its good to use os.sep, may be created in one os and read by another.. maybe check both / & \